.github/workflows/ci.yml

```diff
@@ -157,117 +100,51 @@ jobs:
-            librealm-ios-catalyst
-
   build-android-combined:
     name: Combine Android binaries
-    runs-on: ubuntu-latest
     needs: build-native
+    uses: ./.github/workflows/binary-combine-android.yml
+    with:
+      os: ubuntu
 
-    steps:
-      - name: Fetch x86 build
-        uses: actions/download-artifact@v2
-        with:
-          name: librealm-android-x86
-          path: binary/android
-      - name: Fetch x86_64 build
-        uses: actions/download-artifact@v2
-        with:
-          name: librealm-android-x86_64
-          path: binary/android
-      - name: Fetch armeabi-v7a build
-        uses: actions/download-artifact@v2
-        with:
-          name: librealm-android-armeabi-v7a
-          path: binary/android
-      - name: Fetch arm64-v8a build
-        uses: actions/download-artifact@v2
-        with:
-          name: librealm-android-arm64-v8a
-          path: binary/android
-
-      - name: Store combined artifact
-        uses: actions/upload-artifact@v2
-        with:
-          name: librealm-android
-          path: binary/android
-          retention-days: 1
-
-      - name: Delete individual build artifacts
-        uses: geekyeggo/delete-artifact@v1
-        with:
-          name: |
-            librealm-android-x86
-            librealm-android-x86_64
-            librealm-android-armeabi-v7a
-            librealm-android-arm64-v8a
+  build-ios-xcframework:
+    name: Combine IOS binaries
+    needs: build-native
+    uses: ./.github/workflows/binary-combine-ios.yml
+    with:
+      os: macos
 
-# Dart jobs
+# Linux jobs
 
-  tests-dart:
-    runs-on: ${{ matrix.platform == 'linux' && 'ubuntu' || matrix.platform }}-${{ matrix.architecture || 'latest' }}
-    name: Tests ${{ matrix.platform }} ${{ matrix.architecture }}
+  dart-tests:
     strategy:
       fail-fast: false
       matrix:
         include:
-          - platform: macos
-          - platform: macos
+          - os: macos
+          - os: macos
             architecture: arm
-          - platform: windows
-          - platform: linux
+          - os: windows
+          - os: linux
+    runs-on: ${{ matrix.os == 'linux' && 'ubuntu' || matrix.os }}-${{ matrix.architecture || 'latest' }}
+    name: Dart tests on ${{matrix.os }} ${{ matrix.architecture }}
     env:
       BAAS_CLUSTER: ${{ needs.deploy-cluster.outputs.clusterName }}
-      BAAS_DIFFERENTIATOR: dart-${{ matrix.platform }}${{ matrix.architecture && format('-{0}', matrix.architecture) || '' }}
     needs:
-      - baas-matrix
       - deploy-cluster
       - build-native
     steps:
+
+      - name: Generate unique job id
+        id: generate_job_id
+        if: ${{matrix.os != 'windows'}}
+        run: echo "BAAS_DIFFERENTIATOR=d$RANDOM$100" >> $GITHUB_ENV
+        
+      - name: Generate unique job id on Windows
+        id: generate_job_id_win
+        if: ${{matrix.os == 'windows'}}
+        run: echo "BAAS_DIFFERENTIATOR=d$(Get-Random)" >> $env:GITHUB_ENV
```

Comment:
Discussed offline, we don't need the random here.

Reply:
Deleted